File: navigation/src/main/java/com/example/navigation/stepsEngine/field/ReservationEmailField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class ReservationEmailField extends Field {

    @Override
    public String get(FlowState flowState) {
        return flowState.getReservation().getReservationEmail();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/ReservationEmailField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState

class ReservationEmailField : Field() {
    override operator fun get(flowState: FlowState): String? {
        return flowState.reservation.reservationEmail
    }

    override fun getId(): String {
        return FieldName.RESERVATION_EMAIL
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}